### ISSUE TITLE
[Bug]: nvim-treesitter fails installation on Windows with multiple compilers in path

### ISSUE THREAD

**Issue by Antyos** [LunarVim/LunarVim#1663]
### Problem description

Per @kylo252's recommendation, this is a dedicated thread for the issue I discussed in #1261 where **nvim-treesitter** failed to compile even in a Visual Studio **x64 Native Tools** command prompt. After further investigation on my part, I was able to get everything working, but I still think it is worth its own issue. 

@MarcSchaetz's hunch that I had extra compilers in my path was correct. I had installed Swift/LLVM on my computer and `clang` had made its way into my System-level Path.

I was able to resolve this issues by following the steps in the "Troubleshooting" section of **nvim-treesitter**'s Windows support page: <https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support#troubleshooting>. I added the following line to LunarVim's `init.lua`, substituting MSVC's `cl.exe` in favor of `clang.exe` described on the support page.

```lua
require "nvim-treesitter.install".compilers = { "cl" }
```

> I inserted line between loading the plugins and loading the core. There's definitely a better place for this, I just don't know where that is.

### Suggestions

While I am not intimately familiar with LunarVim (as I have only *just* been able to get it working), here are my suggestions:

1. Set a default compiler for Windows machines that can be overridden in user configuration.

2. Try to find compilers on the user's system and add them to the local path so **nvim-treesitter** can run outside of a developer shell. 

    Even though it's not *too* much of a hassle to run the initial installation in a developer shell, **nvim-treesitter** may need access to a C compiler when installing languages at a later time. At the very least, is it reasonable to add a configuration option to set the path to the preferred compiler?

### LunarVim version

Commit: f61c4ac (Latest from `lang-refactor` branch)

### Neovim version (>= 0.5)

v0.5.0

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

**Comment by kylo252** [comment]
I've had similar issues and solution was to limit the number of installed parsers because some of them are just plain broken. 

Can you remove all of them and test adding just one or two?

**Comment by Antyos** [comment]
Sorry--I'm not sure I understand what you're asking. Do you want me to remove all the nvim-treesitter parsers, then only define one or two of them in the config and try reinstalling them? Or do you want me to remove all my compilers?

If it's a nvim-treesitter package, I know that the "Python" language for sure raised an error without an explicit compiler set; it raised an error when installing via `:TSInstall python`.

**Comment by kylo252** [comment]
Is this still an issue after removing the problematic parsers? maybe we need some sort of guide for making sure a compiler is set and available in general.

**Comment by kylo252** [comment]
The problem is that I have developer mode enabled on my machines, and it's kinda hard on Windows to figure out what's been a default setting in the environment and what's not after a few updates..

I think the best idea generally is going through the VS C++ Build Tools installer, and adding support for both msvc and clang through it.

**Comment by Antyos** [comment]
I'm a bit confused what you're asking--

On the computer I encountered the issues, I have the **Desktop development with C++** option checked in the VS Installer for both VS Community 2019 and VS Build Tools 2019. Having both installed probably isn't a great idea, but I won't have access to that computer to try anything until Monday.

I also have Clang installed for Swift development on that computer. I haven't really done much with Swift, but I used the Windows installer from their website. While the VS C++ compilers aren't in the path unless I launch a VS developer shell, Clang for Swift is in the path by default and I think this is where the issue was coming from.

Are you saying that the only C++ compilers that should ever be in my path are the ones from Visual Studio, and that I should remove all other ones, i.e., Clang via Swift and MinGW? (I don't actually remember if I have MinGW on that machine.)

**Comment by Antyos** [comment]
I used the [nvim-treesitter/nvim-treesitter/wiki/Windows-support](https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support) page to get my initial installation working (as noted in the original post on this issue).

I had to modify LunarVim's `init.lua` with the line below to get Treesitter to install any of the default parsers on LunarVim's first/second launch.

```lua
require "nvim-treesitter.install".compilers = { "cl" }
```

I'm sure there is a better solution than modifying the base `init.lua`, but at the time (and now) I wasn't sure where that modification should be made.

Is it clear that I was successfully able to get Treesitter to install all the necessary parsers by modifying LunarVim's `init.lua` and am now suggesting potential changes to LunarVim's configuration, such that this is not an issue for other users in the future? _(I hope this doesn't come off as aggressive in any way; I want to make sure we are on the same page because it feels like we aren't.)_

I am unsure if the **nvim-treesitter** people plan to improve windows compatibility with multiple compilers or if they consider the configuration change as "good enough". Assuming they consider the lua config change as "good enough", I suggest that LunarVim implement some method to handle multiple compilers, such as:

```lua
-- config.lua
lvim.builtins.treesitter.cc_compiler = { 'cl' }
```

As LunarVim's official handle for the multiple compiler selection issue.

Alternatively, if `require "nvim-treesitter.install".compilers = { "cl" }` can be placed in `config.lua` without issue (I'm still a noob), then LunarVim may just need to make a note about the necessary configuration if the installation fails.

**Comment by Antyos** [comment]
Just wanted to give a quick update here after installing LunarVim on a clean Windows install--everything is working as expected! In addition to having installed MSVC through VS 2019 Community, I have also installed MinGW and LLVM through Chocolatey and they have not interfered with the build process at all.

I'm thinking the issue I was facing related specifically to the way I installed Swift on my previous system. I don't remember how I installed Swift previously; I thought it was the "normal" way, but I'm thinking it might not have been. If I am able to reproduce the same error in the future, I will update this thread.
